lib/screens/home_screen.dart

```diff
@@ -30,18 +32,59 @@ class HomeScreen extends StatefulWidget {
 }
 
 class _HomeScreenState extends State<HomeScreen> {
-  final int _numPages = 5;
-  final PageController _pageController = PageController(keepPage: true);
-  final ValueNotifier<int> _currentPage = ValueNotifier(0);
-  static const _kPageList = 3;
+  late EaselProvider easelProvider;
+  var cacheManager = GetIt.I.get<LocalDataSource>();
+  final int _numPages = 4;
+  late final PageController _pageController;
 
-  final int _numSteps = 4;
-  final ValueNotifier<int> _currentStep = ValueNotifier(0);
+  late ValueNotifier<int> _currentPage;
 
-  final List pageTitles = [kUploadNFTText, kDescribeNftText, kPriceNftText, kListNftText, ''];
+  late ValueNotifier<int> _currentStep;
+
+  NFT? nft;
+  String? from;
+  final List pageTitles = [kSelectNFTText, kDetailNftText, kPriceNftText, ''];
 
   @override
   void initState() {
+    easelProvider = Provider.of<EaselProvider>(context, listen: false);
+    from = cacheManager.getCacheDynamicType(key: "from");
```

Comment:
Cache manager should come from repository and repository should get it from local data store.